src/porepy/numerics/time_step_control.py

```diff
@@ -1,130 +1,235 @@
-from typing import Tuple, Union
+from __future__ import annotations
+
+import warnings
+from typing import Optional, Union
+
+import numpy as np
 
 __all__ = ["TimeSteppingControl"]
 
 
 class TimeSteppingControl:
-    """Parent class for iteration-based time stepping control routine."""
+    """Parent class for iteration-based time-stepping control routine.
+
+    Parameters:
+        schedule: List containing the target times for the simulation.
+            Unless a constant time step is prescribed, the time-stepping algorithm will adapt
+            the time step so that the scheduled times are guaranteed to be hit.
+
+            The `schedule` list must contain minimally two items, corresponding to the
+            initial and final simulation times. Lists of length > 2 must contain strictly
+            increasing times. Examples of VALID inputs are: [0, 1], [0, 10, 30, 50], and
+            [0, 1*pp.HOUR, 3*pp.HOUR]. Examples of INVALID inputs are: [1], [1, 0], and
+            [0, 1, 1, 2].
+
+            If a constant time step is used (`constant_dt = True`), then the time step
+            (`dt_init`) is required to be compatible with the scheduled times in `schedule`.
+            Otherwise, an error will be raised. Examples of VALID inputs for `constant_dt =
+            True` and `dt_init = 2` are: [0, 2] and [0, 4, 6, 10]. Examples of INVALID inputs
+            for `constant_dt = True` and `dt_init = 2` are [0, 3] and [0, 4, 5, 10].
```

Comment:
Sounds like trouble to me. Why not ignoring the schedule when choosing `constant_dt=True`? The `TimeSteppingControl` anyhow will visit all of the scheduled times, you are currently requesting.

Reply:
The only requirement which one could stick to is that max time - min time is a multiple of `dt_init`.

Reply:
I understand the logic, yes. Sounds like a better and less confusing way to handle constant time steps. So, essentially, we will only need to check if `mod((final_time - initial_time), dt_init) == 0`. Right?

Reply:
We decided to keep it as it is, meaning , to require strictly increasing times.

Reply:
We decided to keep it as it is for the meantime. If, in practice, this becomes too restrictive, we'll go for this alternative.